File: javascript/rna-transcription/rna-transcription.js
var dnaToRna = {
  G: 'C',
  C: 'G',
  T: 'A',
  A: 'U'
}

var dnaTranscriber = function() {};

dnaTranscriber.prototype.toRna = function(dna) {
  return dna.split('')
            .map(function(nucleotide) {
              return dnaToRna[nucleotide];
            }).join('');
}

module.exports = dnaTranscriber;

File: kotlin/rna-transcription/src/main/kotlin/RnaTranscription.kt
val DNA_TO_RNA = mapOf(
    'G' to 'C',
    'C' to 'G',
    'T' to 'A',
    'A' to 'U'
)

fun transcribeToRna(dna: String): String {
    return dna.map { nucleotide -> DNA_TO_RNA[nucleotide] }.joinToString("")
}
